rate.go

```diff
@@ -40,3 +45,22 @@ func (c *Client) GetRateWithContext(ctx context.Context, rateID string) (out *Ra
 	err = c.get(ctx, "rates/"+rateID, &out)
 	return
 }
+
+type getShipmentSmartratesResponse struct {
+	Smartrate *[]*Smartrate `json:"result,omitempty"`
+}
+
```

Comment:
do this inside the `GetShipmentSmartRates` like all the other client funcs: https://github.com/EasyPost/easypost-go/blob/master/shipment.go#L180

Reply:
We also have precedence to do this outside throughout the CL, is one better than the other? If so, what makes it more advantageous than the alternative? 

eg: https://github.com/EasyPost/easypost-go/blob/master/shipment.go#L234-L237

Reply:
heh there are just inconsistencies in this library.

now that I'm looking further, the `XXXWithContext` functions seems to be the reason for these inconsistencies and are not setup ideally.

Here's how are we should approach this going forward:
```
func (c *Client) GetShipmentSmartrates(shipmentID string) (out []*Rate, err error) {
	return c.GetShipmentSmartratesWithContext(nil, shipmentID)
}

func (c *Client) GetShipmentSmartratesWithContext(ctx context.Context, shipmentID string) (out []*Rate, err error) {
        res :=  struct {
	       Smartrates *[]*Rate `json:"result,omitempty"`
        }{Smartrates: &out}
	err = c.get(ctx, "shipments/"+shipmentID+"/smartrate", &res)
	return
}
```

Reply:
we can clean up the rest of the func implementations to follow the above convention in a followup

Reply:
Aha, good to know. I've updated this and will file an issue to correct these.